realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmObjectExtensions.kt

```diff
@@ -56,94 +54,25 @@ import kotlinx.coroutines.flow.flowOf
  */
 @Beta
-fun <T : RealmModel> T.toFlow(): Flow<T> {
-    val obj = this
-    return if (obj is RealmObjectProxy) {
-        val proxy = obj as RealmObjectProxy
-        val realm = proxy.`realmGet$proxyState`().`realm$realm`
-
-        when (realm) {
-            is Realm -> flowFromRealm<T>(realm, obj)
-            is DynamicRealm -> {
-                val dynamicObject = obj as DynamicRealmObject
-                flowFromDynamicRealm(realm, dynamicObject) as Flow<T>
-            }
-            else -> throw UnsupportedOperationException("${realm.javaClass} does not support RxJava. See https://realm.io/docs/java/latest/#rxjava for more details.")
-        }
-    } else {
-        return flowOf(this)
-    }
-}
-
-private fun <T : RealmModel> flowFromRealm(realm: Realm, obj: T): Flow<T> {
-    // Return "as is" if frozen, there will be no listening for changes
-    if (realm.isFrozen) {
-        return flowOf(obj)
-    }
-
-    val config = realm.configuration
-
-    return callbackFlow<T> {
-        // Do nothing if the object is invalid
-        if (!obj.isValid()) {
-            return@callbackFlow
-        }
-
-        // Get instance to ensure the Realm is open for as long as we are listening
-        val flowRealm = Realm.getInstance(config)
-        val listener = RealmChangeListener<T> { listenerObj ->
-            offer(listenerObj.freeze())
-        }
-
-        obj.addChangeListener(listener)
-
-        // Emit current (frozen) value
-        offer(freeze(obj))
-
-        awaitClose {
-            // Remove listener and cleanup
-            if (!flowRealm.isClosed) {
-                obj.removeChangeListener(listener)
-                flowRealm.close()
-            }
-        }
-    }
-}
-
-private fun flowFromDynamicRealm(
-        dynamicRealm: DynamicRealm,
-        dynamicObject: DynamicRealmObject
-): Flow<DynamicRealmObject> {
-    // Return "as is" if frozen, there will be no listening for changes
-    if (dynamicRealm.isFrozen) {
-        return flowOf(dynamicObject)
-    }
-
-    val config = dynamicRealm.configuration
-
-    return callbackFlow<DynamicRealmObject> {
-        // Do nothing if the object is invalid
-        if (!dynamicObject.isValid()) {
-            return@callbackFlow
-        }
-
-        // Get instance to ensure the Realm is open for as long as we are listening
-        val flowRealm = Realm.getInstance(config)
-        val listener = RealmChangeListener<DynamicRealmObject> { listenerObj ->
-            offer(listenerObj.freeze())
-        }
-
-        dynamicObject.addChangeListener(listener)
-
-        // Emit current (frozen) value
-        offer(freeze(dynamicObject))
-
-        awaitClose {
-            // Remove listener and cleanup
-            if (!flowRealm.isClosed) {
-                dynamicObject.removeChangeListener(listener)
-                flowRealm.close()
+fun <T : RealmModel> T?.toFlow(): Flow<T?> {
+    // Return flow with object or null flow if this function is called on null
```

Comment:
Can `this` ever be `null` here?

Reply:
Yes, when calling `findFirst().toflow()`. We got feedback from a user and decided to add this here https://github.com/realm/realm-java/issues/7165